### ISSUE TITLE
XFCE: "lock screen before sleep" not working

### ISSUE THREAD

**Issue by davidak** [NixOS/nixpkgs#38492]
## Issue description

this is a security issue

### Steps to reproduce

1. enable "lock screen before sleep" in settings

![screenshot_2018-04-06_02-03-14](https://user-images.githubusercontent.com/91113/38397590-d54667ca-393e-11e8-91a8-ef27f9a1858d.png)

2. suspend
3. hit start button
4. you are logged in directly

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.32, NixOS, 18.03.131768.a74969256b0 (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0`
 - channels(davidak): `""`
 - channels(root): `"nixos-18.03, nixos-hardware, unstable-18.09pre133640.ea145b68a01"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by davidak** [comment]
The setting uses `xflock4`.

```
[davidak@X230:~]$ xflock4 
Property "/general/LockCommand" does not exist on channel "xfce4-session".
```

**Comment by yegortimoshenko** [comment]
`lightlocker` is not very light, perhaps `slock` would be a good default (it's very small, see  https://github.com/slopjong/slock). I too would support adding something as a fallback to Xfce closure.

**Comment by davidak** [comment]
This also works now again. Was anything changed?

**Comment by michalrus** [comment]
@volth thank you for the kind words! That module has a lot of things hard-coded, though. :( I’ll add a note to maybe clean it up & contribute.

**Comment by yegortimoshenko** [comment]
Not really. You probably have some of the executables that it takes on PATH. There is still no default.

**Comment by davidak** [comment]
@yegortimoshenko you are right. i installed lightlocker for the user and locking works.

**Comment by davidak** [comment]
Same issue in GNOME. "Screen Lock" is "On", but it don't lock when suspending.

![screenshot from 2018-05-09 21-38-16](https://user-images.githubusercontent.com/91113/39839586-d32ab64a-53dc-11e8-886d-2a3741712207.png)

Reproduced with:
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.39, NixOS, 18.03.132275.2352d46904d (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.1`
 - channels(root): `"nixos-18.03.132275.2352d46904d"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by davidak** [comment]
Locking works in GNOME when GDM is used instead of lightdm.

**Comment by alexeymuranov** [comment]
As far as I understand, locking the screen and locking the login keychain is not the same thing, and the second is more important.  Is the keychain getting locked in GNOME with GDM on suspend?

**Comment by transgirllucy** [comment]
is this fixed yet?
